### PR TITLE
Add --dest-precompute-digests option for docker

### PR DESCRIPTION
**Pull request by paulnivin**
This ensures layers are not uploaded that already exist on the
destination registry, in exchange for streaming layers to temporary
files when digests are unknown (ex. compressing "on the fly").

Signed-off-by: Paul Fisher <pfisher@lyft.com>

### CODE REVIEW

File: cmd/skopeo/utils.go [hunk 1]
```diff
@@ -264,6 +266,7 @@ func (opts *imageDestOptions) newSystemContext() (*types.SystemContext, error) {
 	if opts.compressionLevel.present {
 		ctx.CompressionLevel = &opts.compressionLevel.value
 	}
+	ctx.DockerRegistryPushPrecomputeDigests = opts.precomputeDigests
```
Comment (mtrmac) [inline cmd/skopeo/utils.go:269]: Please update `TestImageDestOptionsNewSystemContext`.

File: docs/skopeo-copy.1.md [hunk 2]
```diff
@@ -160,6 +160,10 @@ Bearer token for accessing the source registry.
 
 Bearer token for accessing the destination registry.
 
+**--dest-precompute-digests** _bool-value_
+
+Precompute digests to ensure layers are not uploaded that already exist on the destination registry. Layers with initially unknown digests (ex. compressing "on the fly") will be temporarily streamed to disk.
+
```
Comment (mtrmac) [inline docs/skopeo-copy.1.md:166]: Please update also `completions/bash/skopeo`.

**Comment by mtrmac** [comment]
> This PR pulls in tip of main on containers/image for [containers/image#1393](https://github.com/containers/image/pull/1393). I can rebase this PR after skopeo gets bumped to a new containers/image release.

(This can be merged without waiting for a full c/image release.)

File: docs/skopeo-copy.1.md (hunk 2 above)
Comment (paulnivin) [reply docs/skopeo-copy.1.md:166]: Done

**Pull request merged**